presto-main/src/main/java/com/facebook/presto/operator/TopNRowNumberOperator.java

```diff
@@ -339,7 +338,7 @@ private Page getPage()
             return null;
         }
         Page page = pageBuilder.build();
-        localUserMemoryContext.addBytes(-sizeDelta);
+        localUserMemoryContext.setBytes(localUserMemoryContext.getBytes() - sizeDelta);
```

Comment:
For this case, I acknowledge that you actually want delta value． Let's talk about this.